### PR TITLE
Convert RSA tests to new framework

### PR DESCRIPTION
**Pull request by jon-oracle**
- [x] tests are added or updated

Tests converted to use new functions.

### CODE REVIEW

File: test/rsa_test.c [hunk 1]
```diff
@@ -232,109 +234,115 @@ int main(int argc, char *argv[])
...
+    unsigned char ptext[256];
+    unsigned char ctext[256];
+    static unsigned char ptext_ex[] = "\x54\x85\x9b\x34\x2c\x49\xea\x2a";
+    unsigned char ctext_ex[256];
+    int plen;
+    int clen = 0;
+    int num;
+    int n;
 
     RAND_seed(rnd_seed, sizeof rnd_seed); /* or OAEP may fail */
```
Comment (mattcaswell) [inline test/rsa_test.c]: Really? Shouldn't be needed.

File: test/rsa_test.c [hunk 2]
```diff
@@ -232,109 +234,115 @@ int main(int argc, char *argv[])
...
+    case 0:
+        clen = key1(key, ctext_ex);
+        break;
+    case 1:
+        clen = key2(key, ctext_ex);
+        break;
+    case 2:
+        clen = key3(key, ctext_ex);
+        break;
+    }
```
Comment (mattcaswell) [inline test/rsa_test.c]: Perhaps this could be a function rather than repeating the same code as above

File: test/rsa_test.c (hunk 1 above)
Comment (jon-oracle) [inline test/rsa_test.c]: Yes seems to work fine without it - removed!

File: test/rsa_test.c (hunk 2 above)
Comment (jon-oracle) [inline test/rsa_test.c]: Done

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**